docs/DeveloperGuide.md

```diff
@@ -1,38 +1,461 @@
+| v2.0    | user     | find my products' past transactions            | keep track of the changes in my products' quantity                                            |
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+* Any mainstream OS with Java `11` installed
 
 ## Glossary
 
-* *glossary item* - Definition
+* *PID (Product ID)* - A unique number assigned to each product for identification purposes.
+* Mainstream OS: Windows, Linux, Unix, MacOS
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Instructions for manual testing**
+
+### Adding a Product
+1. No prerequisites needed.
+
+2. Test case: `new n/Drinking Cup q/20`<br>
+   Expected: The product will be added. Name of the product is `Drinking Cup`, 
+   Quantity of chocolate Milk stock is `20` units.
+
+3. Test case: `new n/Chocolate Milk q/100 p/2.00 d/Marigold HL Milk`<br>
+   Expected: The product will be added. Name of the product is `Chocolate Milk`,
+   Quantity of chocolate Milk stock is `100 ` units
+   Price of each unit is $`2.00`
+   Description of the Chocolate Milk product is `Marigold HL Milk`, which is the brand.
+
+### Editing Product Details
+1. Prerequisites: List all products using `list` command. There should be at least multiple products in the list.
+
+2. Test case: `edit 1 n/Updated name d/Updated description`<br> 
+   Expected: The name and description of the product with Product ID (PID) 1 
+   will be changed to `Updated name` and `Updated description` respectively.
+
+3. Test case: `edit 1 q/100 p/0.99`<br>
+   Expected: The quantity and price of the product with Product ID (PID) 1
+   will be changed to `100` and `0.99` respectively.
+
+### Deleting a product
+1. Prerequisites: List all products using `list` command. There should be a particular product with `pid` of 1 and no 
+product with `pid` of 2.
+
+2. Test case 1: `delete 1`. 
+   Expected: `product` with `pid` of 1 is deleted from the list. `"Product has been deleted"` is printed to the user.
+
+   Test case 2: `delete 2`.
+   Expected: `"Product with pid: 2 not found"`
+
+
+### Finding a keyword in the Product list 
+1. No prerequisites needed.
+
+2. Test case: `find Cor`<br>
+   Expected: A list of products will be printed out if there is a match, 
+   otherwise `No match found.` will be printed out. 
+
 
-## Instructions for manual testing
+### Finding all past transactions for a particular product in the Product list
+1. Prerequisites: List all products using `list` command. There should be at least multiple products in the list.
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+2. Test case: `history 1`<br>
+   Expected:
+   A list of transactions will be printed out if there is a match,
+   otherwise `No match found.` will be printed out.
```

Comment:
seems like a 404 error when i click subscribe at the bottom of the page

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/61624545/7e43079e-b9ce-4cb7-8a00-a29cb95f8dbc)